test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/EventSourceTests.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.Infrastructure;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.AspNetCore.Testing.xunit;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.FunctionalTests
+{
+    public class EventSourceTests
+    {
+        [Fact]
+        public Task LogsConnection()
+        {
+            return TestEventSource(new ListenOptions(new IPEndPoint(IPAddress.Loopback, 0)));
+        }
+
+        [ConditionalFact]
+        [IPv6SupportedCondition]
+        public Task LogsIpv6Connection()
```

Comment:
When I was playing around with TestServer, I noticed that TestServer always assumes IPv4. I wrote this to validate TestServer used ipv6. This test isn't really necessary so I'll back it out, but I'd like to keep the TestServer changes.